src/ORM/FieldType/DBGeneratedColumn.php

```diff
@@ -0,0 +1,288 @@
+    {
+        $this->getChildField()->setTable($tableName);
+        return parent::setTable($tableName);
+    }
+
+    public function hasValue(string $field, array $arguments = [], bool $cache = true): bool
+    {
+        return $this->getChildField()->hasValue($field, $arguments, $cache);
+    }
+
+    public function nullValue(): mixed
+    {
+        return $this->getChildField()->nullValue();
+    }
+
+    public function exists(): bool
+    {
+        return $this->getChildField()->exists();
+    }
+
+    public function obj(string $fieldName, array $arguments = [], bool $cache = false): ?object
+    {
+        return $this->getChildField()->obj($fieldName, $arguments, $cache);
+    }
+
+    public function castingHelper(string $field): ?string
+    {
+        return $this->getChildField()->castingHelper($field);
+    }
+
+    public function __toString(): string
+    {
+        return $this->getChildField()->__toString();
+    }
+
+    public function forTemplate(): string
+    {
+        return $this->getChildField()->forTemplate();
+    }
+
+    public function HTMLATT(): string
+    {
+        return $this->getChildField()->HTMLATT();
+    }
+
+    public function URLATT(): string
+    {
+        return $this->getChildField()->URLATT();
+    }
+
+    public function RAWURLATT(): string
+    {
+        return $this->getChildField()->RAWURLATT();
+    }
+
+    public function ATT(): string
+    {
+        return $this->getChildField()->ATT();
+    }
+
+    public function RAW(): mixed
+    {
+        return $this->getChildField()->RAW();
+    }
+
+    public function JS(): string
+    {
+        return $this->getChildField()->JS();
+    }
+
+    public function JSON(): string
+    {
+        return $this->getChildField()->JSON();
+    }
+
+    public function HTML(): string
+    {
+        return $this->getChildField()->HTML();
+    }
+
+    public function XML(): string
+    {
+        return $this->getChildField()->XML();
+    }
+
+    public function CDATA(): string
+    {
+        return $this->getChildField()->CDATA();
+    }
+
+    public function debug(): string
+    {
+        return $this->getChildField()->debug();
+    }
+
+    public function scalarValueOnly(): bool
+    {
+        return $this->getChildField()->scalarValueOnly();
+    }
+
+    public function renderWith($template, ModelData|array $customFields = []): DBHTMLText
+    {
+        return $this->getChildField()->renderWith($template, $customFields);
+    }
+
+    public function validate(): ValidationResult
+    {
+        return $this->getChildField()->validate();
+    }
+
+    public function writeToManipulation(array &$manipulation): void
+    {
+        // no-op - generated columns can't be updated manually.
+    }
+
+    private function validateChildFieldClass(): void
+    {
+        foreach (static::config()->get('invalid_child_field_types') as $invalidClass) {
+            if (is_a($this->childField, $invalidClass)) {
```

Comment:
is_a() :1st_place_medal:

Reply:
Has to be `is_a()` when dealing with a string. `instanceof` only works with explicit class references.
i.e. `$this->childField instanceof My\Explicit\Class` works but `$this->childField instanceof 'My\Explicit\Class'` does not.

Otherwise I'd have probably used `instanceof` lol